bsp/esp_bsp_devkit/src/esp_bsp_devkit.c

```diff
@@ -16,6 +16,17 @@
 
 static const char *TAG = "BSP-devkit";
 
+#define bsp_btn_cfg(n)     bsp_btn_cfg_##n
+#define BUTTON_TYPE_ADC(n)  CONFIG_BSP_BUTTON_##n##_TYPE_ADC
+#define BUTTON_TYPE_GPIO(n) CONFIG_BSP_BUTTON_##n##_TYPE_GPIO
+
+#define BSP_IOT_BUTTON_CREATE_INTERNAL(i) \
```

Comment:
Is it possible to make it as runtime function? 
`static esp_err_t bsp_iot_button_create_internal(int i){ ... }`

Reply:
Can't,

Because in BSP_IOT_BUTTON_CREATE_INTERNAL  macro definition, have another macro definition bsp_btn_cfg(n)